### PR TITLE
Unit tests for loading python check

### PR DESCRIPTION
**Pull request by remicalixte**
### What does this PR do?

Add more unit tests for loading python check

Scenarios tested:

- [x] Load multiple checks
- [x] Handle loading error
- [x] Handle missing `__version__`
- [x] Error handling while OS thread is locked
- [x] Handle `check()` error
- [x] Handle missing check

### CODE REVIEW

File: pkg/collector/py/loader_test.go [hunk 1]
```diff
@@ -38,13 +39,76 @@ func TestLoad(t *testing.T) {
...
+	// the python module has a version
+	config = integration.Config{Name: "version"}
+	config.Instances = append(config.Instances, []byte("foo: bar"))
+	instances, err = l.Load(config)
+	require.Nil(t, err)
+	assert.Equal(t, 1, len(instances))
+	assert.Equal(t, "1.0.0", instances[0].Version())
+}
+
+func TestLoadVersionLock(t *testing.T) {
```
Comment (hush-hush) [inline pkg/collector/py/loader_test.go]: Could you add a quick comment explaining why we lock OS thread

File: pkg/collector/py/loader_test.go [hunk 2]
```diff
@@ -38,13 +39,76 @@ func TestLoad(t *testing.T) {
...
+	assert.NotNil(t, err)
+
+	// the python module loads and check runs
+	config = integration.Config{Name: "working"}
+	config.Instances = append(config.Instances, []byte("foo: bar"))
+	instances, err = l.Load(config)
+	require.Nil(t, err)
+	assert.Equal(t, 1, len(instances))
+	err = instances[0].Run()
+	fmt.Println(err)
```
Comment (hush-hush) [inline pkg/collector/py/loader_test.go]: debug line ?

File: pkg/collector/py/loader_test.go [hunk 3]
```diff
@@ -38,13 +39,76 @@ func TestLoad(t *testing.T) {
...
+	config = integration.Config{Name: "version"}
+	config.Instances = append(config.Instances, []byte("foo: bar"))
+	instances, err = l.Load(config)
+	require.Nil(t, err)
+	assert.Equal(t, 1, len(instances))
+	assert.Equal(t, "1.0.0", instances[0].Version())
+}
+
+func TestLoadVersionLock(t *testing.T) {
+	glock := newStickyLock()
```
Comment (hush-hush) [inline pkg/collector/py/loader_test.go]: The StickyLock also acquire the GIL which might impact our python code. Could we only lock the OS thread with `runtime.LockOSThread()` here to force python call on the same thread.

**Review by hush-hush — CHANGES_REQUESTED** [review]
Great improvement on the tests, just a few details

File: pkg/collector/py/loader_test.go (hunk 2 above)
Comment (remicalixte) [reply pkg/collector/py/loader_test.go]: yes, I've missed this one

File: pkg/collector/py/loader_test.go (hunk 3 above)
Comment (remicalixte) [reply pkg/collector/py/loader_test.go]: Okay I see

**Review by hush-hush — APPROVED** [review]

**Pull request merged**